src/execution/execute.ts

```diff
@@ -1564,9 +1565,10 @@ function yieldSubsequentPayloads(
   initialResult: ExecutionResult,
 ): AsyncGenerator<AsyncExecutionResult, void, void> {
   let _hasReturnedInitialResult = false;
+  let isDone = false;
 
   function race(): Promise<IteratorResult<AsyncExecutionResult>> {
-    if (exeContext.subsequentPayloads.length === 0) {
+    if (exeContext.subsequentPayloads.length === 0 || isDone) {
```

Comment:
Why was the `isDone` added here? Shouldn't it be `!isDone`?

Reply:
I think this check is in the wrong spot, it should be in the `next()` function not `race`. Pushing up a fix now.